ocp_resources/catalog_source.py

```diff
@@ -1,34 +1,41 @@
-from ocp_resources.constants import TIMEOUT_4MINUTES
 from ocp_resources.resource import NamespacedResource
 
 
 class CatalogSource(NamespacedResource):
+    """
+    CatalogSource 'OLM' API official docs:
+        https://olm.operatorframework.io/docs/concepts/crds/catalogsource/
+    CatalogSource in Openshift docs:
+        https://docs.openshift.com/container-platform/4.9/rest_api/operatorhub_apis/catalogsource-operators-coreos-com-v1alpha1.html
+    """
+
     api_group = NamespacedResource.ApiGroup.OPERATORS_COREOS_COM
 
     def __init__(
         self,
-        name=None,
-        namespace=None,
-        client=None,
         source_type=None,
         image=None,
         display_name=None,
         publisher=None,
-        teardown=True,
-        yaml_file=None,
-        delete_timeout=TIMEOUT_4MINUTES,
         update_strategy_registry_poll_interval=None,
         **kwargs,
     ):
-        super().__init__(
-            name=name,
-            namespace=namespace,
-            client=client,
-            teardown=teardown,
-            yaml_file=yaml_file,
-            delete_timeout=delete_timeout,
-            **kwargs,
-        )
+        """
+        Args:
+            source_type (str): Name of the source type, could be one of the following:
+                grpc with an image reference: OLM pulls the image and runs the pod,
+                    which is expected to serve a compliant API.
+                grpc with an address field: OLM attempts to contact
+                    the gRPC API at the given address. This should not be used in most cases.
```

Comment:
we do not duplicate the api docs; please include the type and short description; in this case `Name of the source type` is enough.

Applies to all places in this PR